src/accelerate/checkpointing.py

```diff
@@ -214,3 +230,322 @@ def load_custom_state(obj, path, index: int = 0):
+        max_shard_size (`int` or `str`, *optional*, defaults to `"10GB"`):
+            The maximum size for a checkpoint before being sharded. Checkpoints shard will then be each of size lower
+            than this size. If expressed as a string, needs to be digits followed by a unit (like `"5MB"`).
+
+            <Tip warning={true}>
+
+            If a single weight of the model is bigger than `max_shard_size`, it will be in its own checkpoint shard
+            which will be bigger than `max_shard_size`.
+
+            </Tip>
+
+        safe_serialization (`bool`, *optional*, defaults to `False`):
+            Whether to save the model using `safetensors` or the traditional PyTorch way (that uses `pickle`).
+        variant (`str`, *optional*):
+            If specified, weights are saved in the format pytorch_model.<variant>.bin.
+        keys_to_ignore_on_save (`List[str]`, *optional*):
+            The list of key that should not be saved
+    """
+
+    if safe_serialization and not is_safetensors_available():
+        raise ImportError("`safe_serialization` requires the `safetensors library: `pip install safetensors`.")
+
+    if os.path.isfile(save_directory):
+        logger_simple.error(f"Provided path ({save_directory}) should be a directory, not a file")
+        return
+
+    os.makedirs(save_directory, exist_ok=True)
+
+    # Only save the model itself if we are using distributed training
+    model_to_save = unwrap_model(model)
+
+    # Save the model
+    if state_dict is None:
+        state_dict = model_to_save.state_dict()
+
+    # Handle the case where some state_dict keys shouldn't be saved
+    if keys_to_ignore_on_save is not None:
+        for ignore_key in keys_to_ignore_on_save:
+            if ignore_key in state_dict.keys():
+                del state_dict[ignore_key]
+    if safe_serialization:
+        # Safetensors does not allow tensor aliasing.
+        # We're going to remove aliases before saving
+        ptrs = collections.defaultdict(list)
+        for name, tensor in state_dict.items():
+            ptrs[id_tensor_storage(tensor)].append(name)
+
+        # These are all the pointers of shared tensors.
+        shared_ptrs = {ptr: names for ptr, names in ptrs.items() if len(names) > 1}
+        warn_names = set()
+        for names in shared_ptrs.values():
+            # Removing the keys which are declared as known duplicates on
+            # load. This allows to make sure the name which is kept is consistent.
+
+            keys_to_ignore_on_load_missing = None
+            if keys_to_ignore_on_load_missing is not None:
+                found = 0
+                for name in sorted(names):
+                    matches_pattern = any(re.search(pat, name) for pat in keys_to_ignore_on_load_missing)
+                    if matches_pattern and name in state_dict:
+                        found += 1
+                        if found < len(names):
+                            del state_dict[name]
```

Comment:
This part should be removed here, this is specific to Transformers.

Reply:
Fixed [8d853b4](https://github.com/huggingface/accelerate/pull/1641/commits/8d853b4c7918360b91877737c363b5722b4c8e48)